docs/guides/caching.md

```diff
@@ -0,0 +1,117 @@
+# Caching
+
+When using the recommended approach for initializing experiences with this SDK, the logic will take fetch the experience entry, other entries, assets, and patterns under the hood and return the final experience object.
```

Comment:
[Optional] Also would be nice if we can be explicit about `other entries, assets`. That what these other entries, assets are. So maybe we can mention something about bound entries and assets.